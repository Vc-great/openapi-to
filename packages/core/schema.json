{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "@openapi-to/define-config",
  "title": "JSON schema for @openapi-to/core define-config",
  "type": "object",
  "required": [],
  "properties": {
    "root": {
      "type": "string",
      "description": "Project root directory. Can be an absolute path, or a path relative from the location of the config file itself.",
      "default": "process.cwd()"
    },
    "input": {
      "type": "object",
      "description": "",
      "properties": {
        "path": {
          "type": "string",
          "description": "Path to be used as the input. Can be an absolute path, or a path relative from the defined root option."
        }
      }
    },
    "output": {
      "type": "object",
      "description": "",
      "properties": {
        "path": {
          "type": "string",
          "description": "Path to be used to export all generated files. Can be an absolute path, or a path relative based of the defined root option."
        },
        "clean": {
          "type": "boolean",
          "description": "Remove previous generated files and folders."
        },
        "write": {
          "type": "boolean",
          "description": "Enabled or disable the writing to the filesystem. This is being used for the playground.",
          "default": "true"
        }
      }
    },
    "hooks": {
      "type": "object",
      "description": "Hooks that can be called when a specific action is done in openapi-to. ",
      "properties": {
        "done": {
          "oneOf": [
            {
              "type": "string",
              "description": "Hook that will be called at the end of all executions. Useful for running Prettier or ESLint to use your own linting structure."
            },
            {
              "type": "array",
              "items": {
                "type": "string"
              },
              "description": "Hook that will be called at the end of all executions. Useful for running Prettier or ESLint to use your own linting structure."
            }
          ]
        }
      }
    },
    "plugins": {
      "oneOf": [
        {
          "type": "array",
          "items": {
            "type": "array",
            "prefixItems": [
              {
                "type": "string"
              }
            ],
            "items": {
              "type": "object"
            }
          }
        },
        {
          "type": "object"
        }
      ],
      "description": "Plugins"
    }
  }
}
