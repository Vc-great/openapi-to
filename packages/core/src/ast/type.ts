import type { JSDocTagStructure } from "ts-morph";
import type {
  AmbientableNodeStructure,
  ExportableNodeStructure,
  JSDocableNodeStructure,
  Structure,
} from "ts-morph";
import type {
  BindingNamedNodeStructure,
  ExclamationTokenableNodeStructure,
  KindedStructure,
  OptionalKind,
  StructureKind,
  TypedNodeStructure,
  VariableDeclarationKind,
  WriterFunction,
} from "ts-morph";

export interface VariableStatementStructureOfObject
  extends Structure,
    VariableStatementSpecificStructure,
    JSDocableNodeStructure,
    AmbientableNodeStructure,
    ExportableNodeStructure {}

interface VariableStatementSpecificStructure
  extends KindedStructure<StructureKind.VariableStatement> {
  declarationKind?: VariableDeclarationKind;
  declarations: OptionalKind<VariableDeclarationStructure>[];
}

export interface VariableDeclarationStructure
  extends Structure,
    VariableDeclarationSpecificStructure,
    BindingNamedNodeStructure,
    InitializerExpressionableNodeStructure,
    TypedNodeStructure,
    ExclamationTokenableNodeStructure {}

interface VariableDeclarationSpecificStructure
  extends KindedStructure<StructureKind.VariableDeclaration> {}

export interface InitializerExpressionableNodeStructure {
  initializer?: string | WriterFunction | ObjectStructure[];
}

export interface ObjectStructure {
  key: string;
  value: string;
  docs: JSDocStructure[];
}

export interface JSDocStructure {
  /**
   * The description of the JS doc.
   * @remarks To force this to be multi-line, add a newline to the front of the string.
   */
  description?: string;
  /** JS doc tags (ex. `&#64;param value - Some description.`). */
  tags?: OptionalKind<JSDocTagStructure>[];
}

export type JSDoc = JSDocStructure[];
