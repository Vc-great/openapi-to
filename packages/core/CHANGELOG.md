# @openapi-to/core

## 3.0.1

### Patch Changes

- fix bug

## 3.0.0

### Major Changes

- fix bug
- fix: bug
- update
- refactor

## 3.0.0-alpha.5

### Major Changes

- fix: bug

## 3.0.0-alpha.4

### Major Changes

- fix bug

## 3.0.0-alpha.3

### Major Changes

- refactor

## 3.0.0-alpha.2

### Major Changes

- fix

## 2.4.0-alpha.1

### Patch Changes

- fix bug

## 2.4.0-alpha.0

### Minor Changes

- update all

## 2.3.0

### Minor Changes

- generate swr

## 2.2.0

### Minor Changes

- fix bug

## 2.1.0

### Minor Changes

- optimize request type zod

## 2.0.0

### Major Changes

- [`e519466`](https://github.com/Vc-great/openapi-to/commit/e5194667c7416e817a498d592c357a7ae9c05f22) Thanks [@Vc-great](https://github.com/Vc-great)! - v2

- [`a39a900`](https://github.com/Vc-great/openapi-to/commit/a39a9002dda434d8a65768f55c69875ed8ad1eea) - fix bug

- 2.0

- [`367be25`](https://github.com/Vc-great/openapi-to/commit/367be252aa434487c09c4566e77792839867b509) - fix bug

- [`7fd9e58`](https://github.com/Vc-great/openapi-to/commit/7fd9e58417ef5563dedf945fbb030b70b8b09bd6) - rc

## 2.0.0-rc.5

### Major Changes

- rc

## 2.0.0-alpha.4

### Major Changes

- fix bug

## 2.0.0-alpha.3

### Major Changes

- fix bug

## 2.0.0-alpha.2

### Major Changes

- v2
