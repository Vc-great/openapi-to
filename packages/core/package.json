{
  "name": "@openapi-to/core",
  "version": "2.4.0-alpha.1",
  "description": "Generator core",
  "keywords": [
    "typescript",
    "plugins",
    "codegen"
  ],
  "repository": {
    "type": "git",
    "url": "",
    "directory": "packages/core"
  },
  "license": "MIT",
  "author": "",
  "sideEffects": false,
  "type": "module",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.cjs",
      "default": "./dist/index.cjs"
    },
    "./utils": {
      "import": "./dist/utils.js",
      "require": "./dist/utils.cjs",
      "default": "./dist/utils.cjs"
    },
    "./package.json": "./package.json",
    "./*": "./*"
  },
  "main": "./dist/index.cjs",
  "module": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "files": [
    "src",
    "dist",
    "*.d.ts",
    "*.d.cts",
    "schema.json",
    "!/**/**.test.**",
    "!/**/__tests__/**"
  ],
  "scripts": {
    "build": "tsup",
    "clean": "npx rimraf ./dist",
    "lint": "bun biome lint .",
    "lint:fix": "bun biome lint --write --unsafe .",
    "release": "pnpm publish --no-git-check",
    "release:canary": "bash ../../.github/canary.sh && node ../../scripts/build.js canary && pnpm publish --no-git-check -tag canary",
    "release:alpha": "bash ../../.github/canary.sh && node ../../scripts/build.js alpha && pnpm publish --no-git-check -tag alpha",
    "start": "tsup --watch",
    "test": "vitest --passWithNoTests",
    "typecheck": "tsc -p ./tsconfig.json --noEmit --emitDeclarationOnly false"
  },
  "dependencies": {
    "@types/json-schema": "^7.0.15",
    "axios": "^1.6.7",
    "code-block-writer": "^12.0.0",
    "consola": "^3.2.3",
    "directory-tree": "^3.5.1",
    "do-swagger2openapi": "^7.0.18",
    "find-up": "^7.0.0",
    "fs-extra": "^11.2.0",
    "is-chinese": "^2.0.0",
    "is-url": "^1.2.4",
    "js-runtime": "^0.0.7",
    "json-schema": "^0.4.0",
    "lodash": "^4.17.21",
    "natural-orderby": "^3.0.2",
    "oas": "^24.9.0",
    "picocolors": "^1.0.0",
    "pinyin-pro": "^3.25.0",
    "seedrandom": "^3.0.5",
    "semver": "^7.5.4",
    "tinyrainbow": "^1.2.0",
    "ts-morph": "^23.0.0"
  },
  "devDependencies": {
    "@openapi-to/config-ts": "workspace:*",
    "@openapi-to/config-tsup": "workspace:*",
    "@types/fs-extra": "^11.0.4",
    "@types/is-url": "^1.2.32",
    "@types/lodash": "^4.17.9",
    "@types/seedrandom": "^3.0.8",
    "@types/semver": "^7.5.6",
    "ora": "^7.0.1"
  },
  "engines": {
    "node": ">=18",
    "pnpm": ">=8.3.0"
  },
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org/"
  }
}
