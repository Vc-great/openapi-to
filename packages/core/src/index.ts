export * from "./defineConfig";
export * from "./errors.ts";
export { Generator } from "./Generator.ts";
export { PackageManager } from "./PackageManager.ts";
// dprint-ignore
export { SchemaGenerator } from "./SchemaGenerator.ts";
export * from "./types.ts";
/*
export default build;*/
export { AST } from "./ast/ast.ts";
export { build, requestRemoteData } from "./build.ts";
export { OpenAPI } from "./openapi/OpenAPI.ts";
export { createPlugin } from "./plugins.ts";

/*



export interface _Register {}
export type Plugins = _Register;
export type OptionsPlugins = { [K in keyof Plugins]: Plugins[K]["options"] };
* */
