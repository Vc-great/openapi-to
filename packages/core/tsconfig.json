{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "baseUrl": ".",
    "types": ["vitest/globals", "bun-types"]
  },
  "include": ["src/**/*.ts", "./package.json", "vitest.config.ts", "tsup.config.ts", "../config-ts/reset.d.ts"]
}
