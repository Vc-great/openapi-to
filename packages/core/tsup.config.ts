import { optionsCJS, optionsESM } from "@openapi-to/config-tsup";

import { defineConfig } from "tsup";

export default defineConfig([
  {
    ...optionsCJS,
    noExternal: [/find-up/],
  },
  optionsESM,
  {
    ...optionsCJS,
    entry: {
      utils: "src/utils/index.ts",
    },
    name: "utils",
    noExternal: [/find-up/],
  },
  {
    ...optionsESM,
    entry: {
      utils: "src/utils/index.ts",
    },
    name: "utils",
    noExternal: [/find-up/],
  },
]);
