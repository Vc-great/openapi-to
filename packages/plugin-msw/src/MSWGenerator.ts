import path from 'node:path'

import { URLPath } from '@openapi-to/core/utils'

import _ from 'lodash'
import { VariableDeclarationKind } from 'ts-morph'

import type { HttpMethod, ObjectStructure, PluginContext } from '@openapi-to/core'
import { formatClassName as formatFakerClassName,formatInstanceName, formatFileName as formatFakerNameSpaceFileName } from '@openapi-to/plugin-faker/utils'
import type { Operation } from 'oas/operation'
import type { ImportDeclarationStructure, VariableStatementStructure } from 'ts-morph'
import type { Config } from './types.ts'
import {formatFileName} from "./utils.ts";
import {HANDLER_SUFFIX} from "./constants.ts";
type ImportStatementsOmitKind = Omit<ImportDeclarationStructure, 'kind'>

export class MSWGenerator {
  private operation: Operation | undefined
  private oas: Config['oas']
  private readonly openapi: Config['openapi']
  private readonly ast: Config['ast']
  private readonly pluginConfig: Config['pluginConfig']
  private readonly openapiToSingleConfig: Config['openapiToSingleConfig']
  private handlerCache: Set<string> = new Set<string>()

  constructor({ oas, openapi, ast, pluginConfig, openapiToSingleConfig }: Config) {
    this.oas = oas
    this.ast = ast
    this.pluginConfig = pluginConfig
    this.openapiToSingleConfig = openapiToSingleConfig
    this.openapi = openapi
  }

  get currentTagName() {
    return this.openapi.currentTagNameOfPinYin
  }
  get handlersName(): string {
    return `${this.currentTagName}Handler`
  }

  get fileName(): string {
    return formatFileName(this.currentTagName)
  }

  get fakerInstanceName(): string {
    return formatInstanceName(this.currentTagName)
  }

  get fakerNameSpaceFileName(): string {
    return formatFakerNameSpaceFileName(this.currentTagName)
  }

  build(context: PluginContext): void {
    _.forEach(_.values(this.openapi.pathGroupByTag), (pathGroup) => {
      const handlerStatements = _.map(pathGroup, ({ path, method, tag }) => {
        this.operation = this.openapi.setCurrentOperation(path, method, tag)
        return this.generatorHandler(path, method)
      })

      const filePath = path.resolve(this.openapiToSingleConfig.output.dir, `${this.fileName}.ts`)
      this.handlerCache.add(this.handlersName)

      this.ast.createSourceFile(filePath, {
        statements: [...this.generateImport(), this.generatorHandlers(handlerStatements), ...this.generatorExport()],
      })
    })

    this.generateHandlers()
  }

  /**
   * @example
   * ```
   * import {petHandlers } from'./petHandlers'
   *
   * export const handlers = [petHandlers]
   * ```
   */
  generateHandlers() {
    const importStatements: Array<ImportDeclarationStructure> = _.chain([...this.handlerCache.keys()])
      .map((handlerName) => {
        const name = handlerName.replace(HANDLER_SUFFIX,'')
        return this.ast.generateImportStatements([
          {
            namedImports: [name],
            moduleSpecifier: `./${formatFileName(name)}`,
          },
        ])
      })
      .push(
        this.ast.generateImportStatements([
          {
            namedImports: ['HttpHandler'],
            moduleSpecifier: 'msw',
          },
        ]),
      )
      .flatten()
      .value()

    const exportStatements = this.ast.generateVariableStatements({
      declarationKind: VariableDeclarationKind.Const,
      declarations: [
        {
          name: 'handlers',
          type: 'Array<HttpHandler>',
          initializer: `[${[...this.handlerCache].map((item) => `...${item}`).join(',')}]`,
        },
      ],
      isExported: true,
    })

    const filePath = path.resolve(this.openapiToSingleConfig.output.dir, 'handlers' + '.ts')

    this.ast.createSourceFile(filePath, {
      statements: [...importStatements, exportStatements],
    })
  }

  /**
   * @example
   * ```
   * export const petMSW = handlers
   *                       .filter(x=>x.start)
   *                       .map(x=>x.msw)
   * ```
   */
  generatorExport(): Array<VariableStatementStructure> {
    return [
      this.ast.generateVariableStatements({
        declarationKind: VariableDeclarationKind.Const,
        declarations: [
          {
            name: this.handlersName,
            type: 'Array<HttpHandler>',
            initializer: `handlers
                          .filter(x=>x.start)
                          .map(x=>x.msw)`,
          },
        ],
        isExported: true,
      }),
    ]
  }

  /**
   *
   * @example
   * ```
   * import {HttpResponse,http} from 'msw'
   * import { petFaker } from "./petFaker";
   * ```
   */
  generateImport(): Array<ImportDeclarationStructure> {
    const request: ImportStatementsOmitKind = {
      namedImports: ['HttpResponse', 'http', 'HttpHandler'],
      moduleSpecifier: 'msw',
    }

    const faker: ImportStatementsOmitKind = {
      namedImports: [this.fakerInstanceName],
      moduleSpecifier: `./${this.fakerNameSpaceFileName}`,
    }

    const statements = _.chain([] as Array<ImportStatementsOmitKind>)
      .push(request)
      .push(faker)
      .filter(Boolean)
      .value()

    return this.ast.generateImportStatements(statements)
  }

  /**
   *
   * @example
   * ```
   * {
   *         name:"find",
   *         start:true,
   *         msw:http.get('/pet', (req) => {
   *             return HttpResponse.json([{}])
   *         })
   * }
   * ```
   */
  generatorHandler(path: string, method: HttpMethod): string {
    const objectStructure: Array<ObjectStructure> = [
      {
        key: 'name',
        value: `'${this.openapi.requestName}'`,
      },
      {
        key: 'start',
        value: 'false',
      },
      {
        key: 'msw',
        value: `http.${method}('${new URLPath(path).toURLPath}', (req) => {
                return HttpResponse.json(${this.fakerInstanceName}.${this.openapi.requestName}())
            })`,
      },
    ]
    return this.ast.generateObject$2(objectStructure)
  }

  /**
   *
   * @example
   * ```
   * const handlers = [
   *     {
   *         name:"find",
   *         start:true,
   *         msw:http.get('/pet', (req) => {
   *             return HttpResponse.json([{}])
   *         })
   *     }
   *]
   * ```
   */
  generatorHandlers(objectStructures: Array<string>): VariableStatementStructure {
    return this.ast.generateVariableStatements({
      declarationKind: VariableDeclarationKind.Const,
      isExported: false,
      docs: [{ description: '' }],
      declarations: [
        {
          name: 'handlers',
          initializer: `[${objectStructures.join(',')}]`,
        },
      ],
    })
  }
}
