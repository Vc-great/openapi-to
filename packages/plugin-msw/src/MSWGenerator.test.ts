import path from "node:path";
import { fileURLToPath } from "node:url";

import { AST, OpenAPI } from "@openapi-to/core";

import _ from "lodash";
import Oas from "oas";

import petStore from "../mock/petstore.json";
import { MSWGenerator } from "./MSWGenerator.ts";

import type { OpenapiToSingleConfig } from "@openapi-to/core";
const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

describe("MswGenerator", async () => {
  const openapiToSingleConfig: OpenapiToSingleConfig = {
    name: "",
    root: "",
    input: {
      path: "",
    },
    output: {
      dir: path.resolve(__dirname, "../gen"),
    },
    plugins: [],
    pluginNames: [],
  };

  const context = {
    output: "",
  };
  const ast = new AST();
  // @ts-expect-error Not a canonical document
  const oas = new Oas(petStore);
  const openapi = new OpenAPI({}, oas);
  const pluginConfig = {};
  test("MSWGenerator getFullText", () => {
    const requestGenerator = new MSWGenerator({
      oas,
      ast,
      openapi,
      pluginConfig,
      openapiToSingleConfig,
    });
    requestGenerator.build(context);
    const text = _.chain(ast.sourceFile)
      .map((sourceFile) => sourceFile.getFullText())
      .join("\n")
      .value();

    expect(text).toMatchSnapshot();
  });
});
