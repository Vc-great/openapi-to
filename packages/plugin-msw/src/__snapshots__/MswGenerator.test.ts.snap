// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MswGenerator > MSWGenerator getFullText 1`] = `
"import { HttpResponse, http, HttpHandler } from "msw";
import { petFakerService } from "./pet-faker.service";
/** */
const handlers = [{
    name: 'testPost',
    start: false,
    msw: http.post('/pet/test', (req) => {
        return HttpResponse.json(petFakerService.testPost())
    })
}, {
    name: 'testPut',
    start: false,
    msw: http.put('/pet/test', (req) => {
        return HttpResponse.json(petFakerService.testPut())
    })
}, {
    name: 'delByTest',
    start: false,
    msw: http.delete('/pet/test', (req) => {
        return HttpResponse.json(petFakerService.delByTest())
    })
}, {
    name: 'testIdGet',
    start: false,
    msw: http.get('/pet/test/:testId', (req) => {
        return HttpResponse.json(petFakerService.testIdGet())
    })
}, {
    name: 'uploadImagePost',
    start: false,
    msw: http.post('/pet/:petId/uploadImage', (req) => {
        return HttpResponse.json(petFakerService.uploadImagePost())
    })
}, {
    name: 'create',
    start: false,
    msw: http.post('/pet', (req) => {
        return HttpResponse.json(petFakerService.create())
    })
}, {
    name: 'update',
    start: false,
    msw: http.put('/pet', (req) => {
        return HttpResponse.json(petFakerService.update())
    })
}, {
    name: 'findByStatusGet',
    start: false,
    msw: http.get('/pet/findByStatus', (req) => {
        return HttpResponse.json(petFakerService.findByStatusGet())
    })
}, {
    name: 'findByTagsGet',
    start: false,
    msw: http.get('/pet/findByTags', (req) => {
        return HttpResponse.json(petFakerService.findByTagsGet())
    })
}, {
    name: 'findByPetId',
    start: false,
    msw: http.get('/pet/:petId', (req) => {
        return HttpResponse.json(petFakerService.findByPetId())
    })
}, {
    name: 'petIdPost',
    start: false,
    msw: http.post('/pet/:petId', (req) => {
        return HttpResponse.json(petFakerService.petIdPost())
    })
}, {
    name: 'delByPetId',
    start: false,
    msw: http.delete('/pet/:petId', (req) => {
        return HttpResponse.json(petFakerService.delByPetId())
    })
}];
export const petHandler: Array<HttpHandler> = handlers
    .filter(x => x.start)
    .map(x => x.msw);

import { HttpResponse, http, HttpHandler } from "msw";
import { storeFakerService } from "./store-faker.service";
/** */
const handlers = [{
    name: 'create',
    start: false,
    msw: http.post('/store/order', (req) => {
        return HttpResponse.json(storeFakerService.create())
    })
}, {
    name: 'findByOrderId',
    start: false,
    msw: http.get('/store/order/:orderId', (req) => {
        return HttpResponse.json(storeFakerService.findByOrderId())
    })
}, {
    name: 'delByOrderId',
    start: false,
    msw: http.delete('/store/order/:orderId', (req) => {
        return HttpResponse.json(storeFakerService.delByOrderId())
    })
}, {
    name: 'inventoryGet',
    start: false,
    msw: http.get('/store/inventory', (req) => {
        return HttpResponse.json(storeFakerService.inventoryGet())
    })
}];
export const storeHandler: Array<HttpHandler> = handlers
    .filter(x => x.start)
    .map(x => x.msw);

import { HttpResponse, http, HttpHandler } from "msw";
import { userFakerService } from "./user-faker.service";
/** */
const handlers = [{
    name: 'createWithArrayPost',
    start: false,
    msw: http.post('/user/createWithArray', (req) => {
        return HttpResponse.json(userFakerService.createWithArrayPost())
    })
}, {
    name: 'createWithListPost',
    start: false,
    msw: http.post('/user/createWithList', (req) => {
        return HttpResponse.json(userFakerService.createWithListPost())
    })
}, {
    name: 'findByUsername',
    start: false,
    msw: http.get('/user/:username', (req) => {
        return HttpResponse.json(userFakerService.findByUsername())
    })
}, {
    name: 'usernamePut',
    start: false,
    msw: http.put('/user/:username', (req) => {
        return HttpResponse.json(userFakerService.usernamePut())
    })
}, {
    name: 'delByUsername',
    start: false,
    msw: http.delete('/user/:username', (req) => {
        return HttpResponse.json(userFakerService.delByUsername())
    })
}, {
    name: 'loginGet',
    start: false,
    msw: http.get('/user/login', (req) => {
        return HttpResponse.json(userFakerService.loginGet())
    })
}, {
    name: 'logoutGet',
    start: false,
    msw: http.get('/user/logout', (req) => {
        return HttpResponse.json(userFakerService.logoutGet())
    })
}, {
    name: 'create',
    start: false,
    msw: http.post('/user', (req) => {
        return HttpResponse.json(userFakerService.create())
    })
}];
export const userHandler: Array<HttpHandler> = handlers
    .filter(x => x.start)
    .map(x => x.msw);

import { petHandler } from "./petHandler.handler";
import { storeHandler } from "./storeHandler.handler";
import { userHandler } from "./userHandler.handler";
import { HttpHandler } from "msw";
export const handlers: Array<HttpHandler> = [...petHandler, ...storeHandler, ...userHandler];
"
`;

exports[`MswGenerator > MswGenerator getFullText 1`] = `
"import { HttpResponse, http, HttpHandler } from "msw";
import { petFakerService } from "./pet-faker.service";
/** */
const handlers = [{
    name: 'testPost',
    start: false,
    msw: http.post('/pet/test', (req) => {
        return HttpResponse.json(petFakerService.testPost())
    })
}, {
    name: 'testPut',
    start: false,
    msw: http.put('/pet/test', (req) => {
        return HttpResponse.json(petFakerService.testPut())
    })
}, {
    name: 'delByTest',
    start: false,
    msw: http.delete('/pet/test', (req) => {
        return HttpResponse.json(petFakerService.delByTest())
    })
}, {
    name: 'testIdGet',
    start: false,
    msw: http.get('/pet/test/:testId', (req) => {
        return HttpResponse.json(petFakerService.testIdGet())
    })
}, {
    name: 'uploadImagePost',
    start: false,
    msw: http.post('/pet/:petId/uploadImage', (req) => {
        return HttpResponse.json(petFakerService.uploadImagePost())
    })
}, {
    name: 'create',
    start: false,
    msw: http.post('/pet', (req) => {
        return HttpResponse.json(petFakerService.create())
    })
}, {
    name: 'update',
    start: false,
    msw: http.put('/pet', (req) => {
        return HttpResponse.json(petFakerService.update())
    })
}, {
    name: 'findByStatusGet',
    start: false,
    msw: http.get('/pet/findByStatus', (req) => {
        return HttpResponse.json(petFakerService.findByStatusGet())
    })
}, {
    name: 'findByTagsGet',
    start: false,
    msw: http.get('/pet/findByTags', (req) => {
        return HttpResponse.json(petFakerService.findByTagsGet())
    })
}, {
    name: 'findByPetId',
    start: false,
    msw: http.get('/pet/:petId', (req) => {
        return HttpResponse.json(petFakerService.findByPetId())
    })
}, {
    name: 'petIdPost',
    start: false,
    msw: http.post('/pet/:petId', (req) => {
        return HttpResponse.json(petFakerService.petIdPost())
    })
}, {
    name: 'delByPetId',
    start: false,
    msw: http.delete('/pet/:petId', (req) => {
        return HttpResponse.json(petFakerService.delByPetId())
    })
}];
export const petHandler: Array<HttpHandler> = handlers
    .filter(x => x.start)
    .map(x => x.msw);

import { HttpResponse, http, HttpHandler } from "msw";
import { storeFakerService } from "./store-faker.service";
/** */
const handlers = [{
    name: 'create',
    start: false,
    msw: http.post('/store/order', (req) => {
        return HttpResponse.json(storeFakerService.create())
    })
}, {
    name: 'findByOrderId',
    start: false,
    msw: http.get('/store/order/:orderId', (req) => {
        return HttpResponse.json(storeFakerService.findByOrderId())
    })
}, {
    name: 'delByOrderId',
    start: false,
    msw: http.delete('/store/order/:orderId', (req) => {
        return HttpResponse.json(storeFakerService.delByOrderId())
    })
}, {
    name: 'inventoryGet',
    start: false,
    msw: http.get('/store/inventory', (req) => {
        return HttpResponse.json(storeFakerService.inventoryGet())
    })
}];
export const storeHandler: Array<HttpHandler> = handlers
    .filter(x => x.start)
    .map(x => x.msw);

import { HttpResponse, http, HttpHandler } from "msw";
import { userFakerService } from "./user-faker.service";
/** */
const handlers = [{
    name: 'createWithArrayPost',
    start: false,
    msw: http.post('/user/createWithArray', (req) => {
        return HttpResponse.json(userFakerService.createWithArrayPost())
    })
}, {
    name: 'createWithListPost',
    start: false,
    msw: http.post('/user/createWithList', (req) => {
        return HttpResponse.json(userFakerService.createWithListPost())
    })
}, {
    name: 'findByUsername',
    start: false,
    msw: http.get('/user/:username', (req) => {
        return HttpResponse.json(userFakerService.findByUsername())
    })
}, {
    name: 'usernamePut',
    start: false,
    msw: http.put('/user/:username', (req) => {
        return HttpResponse.json(userFakerService.usernamePut())
    })
}, {
    name: 'delByUsername',
    start: false,
    msw: http.delete('/user/:username', (req) => {
        return HttpResponse.json(userFakerService.delByUsername())
    })
}, {
    name: 'loginGet',
    start: false,
    msw: http.get('/user/login', (req) => {
        return HttpResponse.json(userFakerService.loginGet())
    })
}, {
    name: 'logoutGet',
    start: false,
    msw: http.get('/user/logout', (req) => {
        return HttpResponse.json(userFakerService.logoutGet())
    })
}, {
    name: 'create',
    start: false,
    msw: http.post('/user', (req) => {
        return HttpResponse.json(userFakerService.create())
    })
}];
export const userHandler: Array<HttpHandler> = handlers
    .filter(x => x.start)
    .map(x => x.msw);

import { petHandler } from "./petHandler.handler";
import { storeHandler } from "./storeHandler.handler";
import { userHandler } from "./userHandler.handler";
import { HttpHandler } from "msw";
export const handlers: Array<HttpHandler> = [...petHandler, ...storeHandler, ...userHandler];
"
`;
